internal/service/ecr/pull_through_cache_rule.go

```diff
@@ -50,6 +51,14 @@ func ResourcePullThroughCacheRule() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+			"credential_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringMatch(
+					regexache.MustCompile(`^arn:aws:secretsmanager:[a-zA-Z0-9-:]+:secret:ecr\-pullthroughcache\/[a-zA-Z0-9\/_+=.@-]+$`),
```

Comment:
The `:aws:` portion of the ARN can change depending on if you're using the GovCloud or China regions. It probably would behoove us to not require that the normal `aws` partition be part of the ARN.

Reply:
@thefirstofthe300 Yeah I already notice it. This is just the pattern written in the documentation (and in the API error message). I can change it, but then users will get the error message from AWS, not Terraform. Is it the best way?

https://docs.aws.amazon.com/de_de/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-pullthroughcacherule.html

Reply:
What about this? This would fail if it doesn't match the existing partitions (aws, aws-cn, aws-us-gov)

```suggestion
					regexache.MustCompile(`^arn:aws(\-cn|\-us\-gov)?:secretsmanager:[a-zA-Z0-9-:]+:secret:ecr\-pullthroughcache\/[a-zA-Z0-9\/_+=.@-]+$`),
```

Reply:
You raise a good point. Two china regions and the govcloud regions don't appear to be supported. For now, it's probably better to leave as you have it.